shipment_advice/models/shipment_advice.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ShipmentAdvice(models.Model):
+    _name = "shipment.advice"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Shipment Advice"
+    _order = "arrival_date asc, id desc"
+
+    def _default_warehouse_id(self):
+        wh = self.env.ref("stock.warehouse0", raise_if_not_found=False)
+        return wh.id or False
+
+    name = fields.Char(
+        default="/", copy=False, index=True, required=True, readonly=True
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("confirmed", "Confirmed"),
+            ("in_progress", "In progress"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        string="Status",
+        default="draft",
+    )
+    warehouse_id = fields.Many2one(
+        comodel_name="stock.warehouse",
+        ondelete="cascade",
+        string="Warehouse",
+        required=True,
+        states={"draft": [("readonly", False)]},
+        readonly=True,
+        check_company=True,
+        default=_default_warehouse_id,
```

Comment:
```suggestion
        default=lambda self: self._default_warehouse_id(),
```

and move the default method after fields pls